File: SpringBoot/src/main/java/com/group3s2i/springboot/Model/OrderCustomerItem.java
package com.group3s2i.springboot.Model;

import com.fasterxml.jackson.annotation.*;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import java.io.Serializable;
import java.time.LocalDateTime;

@Getter
@Setter
@ToString
@Entity
@Table(name = "order_customer_items")
public class OrderCustomerItem implements Serializable {

    private static final long serialVersionUID = 1L;

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @Column(name = "item_seq")
    private  int itemSeq;

    @Column(name = "quantity")
    private @NotNull int quantity;


    @Column(name = "vat_code")
    private  Double vatCode;

    @Column(name = "discount_code")
    private  Double discountCode;

    @Column(name = "created_date")
    private LocalDateTime createdDate;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name="order_id", referencedColumnName = "id")
    @JsonIgnoreProperties("order")
    @ToString.Exclude
    @JsonBackReference(value = "m-d")
    private OrderCustomer orderCustomer;

    @OneToOne
    @JoinColumn(name = "product_id", referencedColumnName = "id")
    @JsonProperty("product")
    private Product product;

    public OrderCustomerItem() {
    }
}

File: ecommerce/lib/models/OrderCustomerItem.dart
import 'Product.dart';

class OrderCustomerItem {
  OrderCustomerItem({
    this.id,
    this.itemSeq,
    this.quantity,
    this.unitPrice,
    this.vatCode,
    this.discountCode,
    this.createdDate,
    this.product,
  });

  int id;
  int itemSeq;
  int quantity;
  double unitPrice;
  double vatCode;
  double discountCode;
  DateTime createdDate;
  Product product;

  factory OrderCustomerItem.fromJson(Map<String, dynamic> json) => OrderCustomerItem(
    id: json["id"],
    itemSeq: json["itemSeq"],
    quantity: json["quantity"],
    unitPrice: json["unitPrice"],
    vatCode: json["vatCode"],
    discountCode: json["discountCode"],
    createdDate: DateTime.parse(json["createdDate"]),
    product: Product.fromJson(json["product"]),
  );

  Map<String, dynamic> toJson() => {
    "id": id,
    "itemSeq": itemSeq,
    "quantity": quantity,
    "unitPrice": unitPrice,
    "vatCode": vatCode,
    "discountCode": discountCode,
    "createdDate": createdDate.toIso8601String(),
    "product": product.toJson(),
  };
}
